### ISSUE TITLE
Classification grid shows wrong transcripts due to using startswith

### ISSUE THREAD

**Issue by davmlaw** [SACGF/variantgrid#522]
View details in Rollbar: [https://rollbar.com/jimmy.andrews/VariantGrid/items/3487/](https://rollbar.com/jimmy.andrews/VariantGrid/items/3487/)

```
User Feedback : ACPP gene page displays classifications i nKAT6A and others
```

The classification grid uses "startswith" in order to deal with transcript versions, for instance on ACPP it will do 

```
published_evidence__refseq_transcript_id__value__startswith', 'NM_001099'
```

However this will also match:

* NM_001099922.2(ALG13)
* NM_001099412(KAT6A)

Need to tighten the regex up

HF to SAP

**Comment by davmlaw** [comment]
It took 3.9 seconds to do the startswith, and then 30+ seconds to do the regex:

```
published_evidence__refseq_transcript_id__value__regex=rf"^{transcript_id}(\.\d+)?$"
```

I then made it do the startswith AND regex and it was 4.6 seconds

Might be worth thinking about pre-processing this and add a record for every symbol a classification is regarded as being under, to speed up this query?

**Comment by davmlaw** [comment]
HF to SAP

https://github.com/SACGF/variantgrid/commit/d75bbd86b6291a442174ba0b0f037e3c77f693ab

**Comment by davmlaw** [comment]
Tested master on variantgrid.com: There is a classification against KAT6A, did not appear on ACPP gene page

**Comment by davmlaw** [comment]
This intially failed in SA Path testing as that old code wasn't calling classification_gene_symbol_filter - Manually patched back only lines in this block from from d91ed4c0d95cb22935f59a408334972971e9d66f - "remove code duplication" so we call it
  
This was done under ba649f5e92d6bc6ddc4b1e15f8fbf279942cba22

This exposed another bug, the ```filter().first(symbol=XX)``` which was fixed via 670fd2ca904130431279b3cb327f7dbe5c9a5033

Tested in SAP: Classifications classified in gene show up, those that showed up before due to startswith no longer do

**Comment by davmlaw** [comment]
Tested in prod

No classifications on ACPP
Went to ALG13 - only ALG13 showed up
Went to KAT6A - only KAT6A showed up
